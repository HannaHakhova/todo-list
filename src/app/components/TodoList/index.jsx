import React from 'react';
import {List, Message, Icon} from 'semantic-ui-react';
import {func, bool, instanceOf, arrayOf, string, shape, number} from 'prop-types';
import Todo from 'components/Todo';


const TodoList = ({todos, handleDelete, handleComplete, activeItem, sortByDateAndCompleted, handleDeleteTodoTag, addTodoTag}) => {
    let todosToLists = [];
    switch (activeItem) {
        case 'all': todosToLists = todos; break;
        case 'completed': todosToLists = todos.filter(todo => todo.completed === true); break;
        default: todosToLists = todos.filter(todo => todo.completed === false); break;
    }

    return (
        <div>
            { todosToLists.length !== 0 ?
                <div>
                    <List relaxed>
                        {todosToLists.map(todo => (<List.Item key={todo.id}>
                            <Todo
                                sortByDateAndCompleted={sortByDateAndCompleted}
                                handleComplete={handleComplete}
                                handleDeleteTodoTag={handleDeleteTodoTag}
                                handleDelete={handleDelete}
                                todo={todo}
                                addTodoTag={addTodoTag} />
                        </List.Item>)

                        )}
                    </List>
                </div> : <div>
                    <br />
                    <Message warning attached='bottom' size='big'>
                        <Icon name='warning' />
                            No todos
                    </Message>
                </div>}
        </div>);
};

TodoList.propTypes = {
    todos: arrayOf(shape({
        title: string.isRequired,
        completed: bool.isRequired,
        createdDate: instanceOf(Date).isRequired,
        tags: arrayOf(shape({
            id: string.isRequired,
            title: string.isRequired,
            bindingId: number.isRequired}).isRequired).isRequired
    }).isRequired).isRequired,
    handleDelete: func.isRequired,
    handleComplete: func.isRequired,
    activeItem: string.isRequired,
    sortByDateAndCompleted: func.isRequired,
    handleDeleteTodoTag: func.isRequired,
    addTodoTag: func.isRequired
};
export default TodoList;