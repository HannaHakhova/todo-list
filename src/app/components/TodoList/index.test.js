/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import {shallow} from 'enzyme';
import TodoList from './index';

const wrapper = shallow(<TodoList
    handleDelete={() => {}}
    handleComplete={() => {}}
    sortByDateAndCompleted={() => {}}
    activeItem='all'
    todos={[{
        id: '1',
        title: 'Todo1',
        completed: true,
        createdDate: new Date(2017, 1, 1)
    },
    {
        id: '2',
        title: 'Todo2',
        completed: true,
        createdDate: new Date(2017, 1, 1)
    },
    {
        id: '3',
        title: 'Todo3',
        completed: true,
        createdDate: new Date(2017, 1, 1)
    }]} />);

describe('TodoList component', () => {
    it('should match expected snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});