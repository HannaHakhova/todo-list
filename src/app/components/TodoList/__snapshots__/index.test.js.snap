// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component should match expected snapshot 1`] = `
<div>
  <div>
    <List
      relaxed={true}
    >
      <ListItem
        key="1"
      >
        <Todo
          handleComplete={[Function]}
          handleDelete={[Function]}
          sortByDateAndCompleted={[Function]}
          todo={
            Object {
              "completed": true,
              "createdDate": 2017-01-31T22:00:00.000Z,
              "id": "1",
              "title": "Todo1",
            }
          }
        />
      </ListItem>
      <ListItem
        key="2"
      >
        <Todo
          handleComplete={[Function]}
          handleDelete={[Function]}
          sortByDateAndCompleted={[Function]}
          todo={
            Object {
              "completed": true,
              "createdDate": 2017-01-31T22:00:00.000Z,
              "id": "2",
              "title": "Todo2",
            }
          }
        />
      </ListItem>
      <ListItem
        key="3"
      >
        <Todo
          handleComplete={[Function]}
          handleDelete={[Function]}
          sortByDateAndCompleted={[Function]}
          todo={
            Object {
              "completed": true,
              "createdDate": 2017-01-31T22:00:00.000Z,
              "id": "3",
              "title": "Todo3",
            }
          }
        />
      </ListItem>
    </List>
  </div>
</div>
`;
